src/data_value.rs

```diff
@@ -0,0 +1,70 @@
+use num_bigint::BigInt;
+use pyo3::{
+    types::{PyDict, PyList},
+    IntoPy, PyObject, Python, ToPyObject,
+};
+
+use crate::lazy_index_map::LazyIndexMap;
+
+/// similar to serde `Value` but with int and float split
+// FIXME remove clone, don't want to accidentally duplicate Vecs
+#[derive(Clone, Debug)]
+pub enum DataValue {
+    Null,
+    Bool(bool),
+    Int(i64),
+    BigInt(BigInt),
+    Uint(u64),
+    Float(f64),
+    String(String),
+    Array(Vec<DataValue>),
+    Object(Box<DataObject>),
```

Comment:
why does this need a box?

Reply:
It doesn't need one but it's more performant to have one (reduces the size of the enum -> reduces size of stack)